x/farming/keeper/reward.go

```diff
@@ -185,8 +246,14 @@ func (k Keeper) CalculateRewards(ctx sdk.Context, farmerAcc sdk.AccAddress, stak
 		return sdk.NewDecCoins()
 	}
 
-	starting, _ := k.GetHistoricalRewards(ctx, stakingCoinDenom, staking.StartingEpoch-1)
-	ending, _ := k.GetHistoricalRewards(ctx, stakingCoinDenom, endingEpoch)
+	starting, found := k.GetHistoricalRewards(ctx, stakingCoinDenom, staking.StartingEpoch-1)
+	if !found {
+		panic("historical rewards for starting epoch not found")
+	}
+	ending, found := k.GetHistoricalRewards(ctx, stakingCoinDenom, endingEpoch)
+	if !found {
+		panic("historical rewards for ending epoch not found")
+	}
```

Comment:
Since you add panic here, I am spinning my head around if it is true that there is any case that `HistoricalRewards` is not found. Can you confirm this?

Reply:
It was added to make sure that the calculating logic is correct. I think the panic never happens.